spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDelete.scala

```diff
@@ -77,33 +101,30 @@ object RewriteDelete extends Rule[LogicalPlan] with PredicateHelper with Logging
     PushDownUtils.pushFilters(scanBuilder, normalizedPredicates)
 
     val scan = scanBuilder.build()
-    val scanRelation = DataSourceV2ScanRelation(table, scan, output)
+    val scanRelation = DataSourceV2ScanRelation(table, scan, toOutputAttrs(scan.readSchema(), output))
 
     val scanPlan = scan match {
```

Comment:
I think this block can be simplified a bit.

```
    scan match {
      case filterable: SupportsFileFilter =>
        val matchingFilePlan = buildFileFilterPlan(cond, scanRelation)
        DynamicFileFilter(scanRelation, matchingFilePlan, filterable)
      case _ =>
        scanRelation
    }
```

Reply:
Good catch. Updated.